diff --git a/mypy/test/helpers.py b/mypy/test/helpers.py
index bb97444273..2f28ab1e47 100755
--- a/mypy/test/helpers.py
+++ b/mypy/test/helpers.py
@@ -3,7 +3,7 @@
 import sys
 import time
 
-from typing import List, Dict, Tuple, TypeVar, Callable
+from typing import List, Dict, Tuple, Callable, Any
 
 from mypy import defaults
 from mypy.myunit import AssertionFailure
@@ -286,10 +286,7 @@ def normalize_error_messages(messages: List[str]) -> List[str]:
     return a
 
 
-_T = TypeVar('_T')
-
-
-def retry_on_error(func: Callable[[], _T], max_wait: float = 1.0) -> _T:
+def retry_on_error(func: Callable[[], Any], max_wait: float = 1.0) -> None:
     """Retry callback with exponential backoff when it raises OSError.
 
     If the function still generates an error after max_wait seconds, propagate
@@ -302,7 +299,8 @@ def retry_on_error(func: Callable[[], _T], max_wait: float = 1.0) -> _T:
     wait_time = 0.01
     while True:
         try:
-            return func()
+            func()
+            return
         except OSError:
             wait_time = min(wait_time * 2, t0 + max_wait - time.time())
             if wait_time <= 0.01:
